ipatests/test_integration/test_idp.py

```diff
@@ -315,3 +315,39 @@ def test_idp_backup_restore(self):
             tasks.kinit_admin(self.master)
             self.master.run_command(["rm", "-rf", backup_path])
             self.master.run_command(["ipa", "idp-del", "testidp"])
+
+    def test_ipaidpsub_attribute(self):
+        """
+        Test case to check if ipaIdpSub is correctly created
+        for IdP and non-IdP users
+
+        Related: https://pagure.io/freeipa/issue/9433
+        """
+        idp_user = "idpuser"
+        standard_user = "nonidpuser"
+
+        tasks.kinit_admin(self.master)
+        cmd = ["ipa", "idp-add", "testidp", "--provider=keycloak",
+               "--client-id=ipa_oidc_client", "--org=master",
+               "--base-url={0}:8443/auth".format(self.client.hostname)]
+        self.master.run_command(cmd, stdin_text="{0}\n{0}".format(
+            self.client.config.admin_password))
+        tasks.user_add(self.master, idp_user,
+                       extra_args=["--user-auth-type=idp",
+                                   "--idp-user-id=testuser1@ipa.test",
+                                   "--idp=testidp"])
+        tasks.user_add(self.master, standard_user)
+
+        # Add idp-user-id to non-idp user
+        cmd = ["ipa", "user-mod", standard_user,
+               "--idp-user-id=new.mydomain.test"]
+        list_user = self.master.run_command(cmd).stdout_text
+        assert "External IdP user identifier: new.mydomain.test" in list_user
+
+        # remove IdP and change idp-user-id for IdP user
```

Comment:
This is actually the real meat: if an external idp is removed, any reference to this idp should be removed from user entries thanks to the referential integrity plugin.
But I don't think that it will remove the "idp" auth type from the user entry.